owncloudApp/src/main/java/com/owncloud/android/ui/preview/PreviewVideoActivity.kt

```diff
@@ -0,0 +1,438 @@
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuInflater
+import android.view.MenuItem
+import android.view.View
+import android.view.Window
+import android.view.WindowManager
+import androidx.annotation.OptIn
+import androidx.appcompat.app.AlertDialog
+import androidx.core.view.MenuProvider
+import androidx.core.view.WindowCompat
+import androidx.core.view.WindowInsetsCompat
+import androidx.core.view.WindowInsetsControllerCompat
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
+import androidx.media3.common.PlaybackException
+import androidx.media3.common.Player
+import androidx.media3.common.util.UnstableApi
+import androidx.media3.exoplayer.DefaultLoadControl
+import androidx.media3.exoplayer.ExoPlaybackException
+import androidx.media3.exoplayer.ExoPlayer
+import androidx.media3.exoplayer.source.MediaSource
+import androidx.media3.exoplayer.trackselection.AdaptiveTrackSelection
+import androidx.media3.exoplayer.trackselection.DefaultTrackSelector
+import androidx.media3.ui.PlayerView
+import com.owncloud.android.R
+import com.owncloud.android.databinding.VideoPreviewBinding
+import com.owncloud.android.domain.files.model.OCFile
+import com.owncloud.android.domain.utils.Event
+import com.owncloud.android.extensions.filterMenuOptions
+import com.owncloud.android.extensions.sendDownloadedFilesByShareSheet
+import com.owncloud.android.presentation.authentication.AccountUtils
+import com.owncloud.android.presentation.files.operations.FileOperation.SetFilesAsAvailableOffline
+import com.owncloud.android.presentation.files.operations.FileOperation.UnsetFilesAsAvailableOffline
+import com.owncloud.android.presentation.files.operations.FileOperationsViewModel
+import com.owncloud.android.presentation.files.removefile.RemoveFilesDialogFragment.Companion.newInstance
+import com.owncloud.android.presentation.previews.PreviewVideoViewModel
+import com.owncloud.android.presentation.transfers.TransfersViewModel
+import com.owncloud.android.ui.activity.FileActivity
+import com.owncloud.android.ui.activity.FileDisplayActivity
+import com.owncloud.android.ui.dialog.ConfirmationDialogFragment
+import com.owncloud.android.ui.fragment.FileFragment
+import com.owncloud.android.ui.preview.PrepareVideoPlayerAsyncTask.OnPrepareVideoPlayerTaskListener
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collectLatest
+import kotlinx.coroutines.launch
+import org.koin.androidx.viewmodel.ext.android.viewModel
+import timber.log.Timber
+
+@OptIn(UnstableApi::class)
+class PreviewVideoActivity : FileActivity(), Player.Listener, OnPrepareVideoPlayerTaskListener, FileFragment.ContainerActivity, MenuProvider {
+    private var mAccount: Account? = null
+
+    private lateinit var playerView: PlayerView
+    private var player: ExoPlayer? = null
+    private lateinit var trackSelector: DefaultTrackSelector
+    private var mAutoplay = true
```

Comment:
Actually, this parameter is used to set the player's property `playWhenReady` and when It's true it will start the playback as soon as the player go into playback state `READY` from `BUFFERING` i.e It will start playing the playback.

- Also, when we play/pause from `PlayerView` this property also changes i.e when play -> true and pause -> false.
- So, we also save this property so that If player is initialsed again, It will know whether to keep the playback paused or start playing as soons as state goes into `READY`.

You can also see when we pause the player. It is paused and If we seek the player progress, it does not start the playback even if player state changes from `BUFFERING` (as player is rendering the new frame due to seek bar progress changed)  to `READY` cause this value is false.